docs/quantum_gate.rst

```diff
@@ -1,4 +1,366 @@
+.. _quantum_gates:
+
 Quantum Gates
 =============
 
-contains gate details
\ No newline at end of file
+Gates in OpenQL are the constructs that refer to operations to be executed somehow on the computing platform.
+
+A gate refers to an operation and to zero or more operands.
+
+Gates are organized in circuits as vectors of gates, i.e. linear sequences of gates.
+A circuit defines the operation of a kernel.
+And a program consists of multiple kernels.
+
+Gates can be subdivided in several kinds.
+This is useful in the description of the passes below.
+
+
+First, gates can be subdivided according to where their execution has effect:
+
+- quantum gates; these gates execute in the quantum computing hardware; these gates have at least one qubit as (implicit or explicit) operand; these gates can have classical registers as operand as well and may rely on some execution capability in classical hardware
```

Comment:
Does this distinction relate to classes in the code or is it only a convenient abstraction?

Reply:
Is a convenient abstraction in the documentation.